File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        
        int answer = 0;
        int now = 0;
        
        for(int i=0; i<section.length; i++){
            if(now >= section[i]) continue;
            answer ++;
            now = section[i]+m-1;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.js
function solution(n, m, section) {
    
    var answer = 0;
    var now = 0;
    
    for(let i=0; i<section.length; i++){
        if(now >= section[i]) continue;
        now = section[i]+m-1;
        answer++;
    }
    
    return answer;
}